### PR TITLE
(SIMP-5893) Update ldap_access_order option

### PR DESCRIPTION
**Pull request by trevor-vaughan**
Update the sssd::provider::ldap::ldap_access_order parameter to support
the ppolicy related options that were added in sssd 1.14.0
  - ppolicy
  - pwd_expire_policy_reject
  - pwd_expire_policy_warn
  - pwd_expire_policy_renew

This has the added benefit of now locking out accounts that have an SSH
key but have been locked out due to invalid passwords

SIMP-5893 #close

### CODE REVIEW

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-sssd/pipelines/41939295

File: functions/ldap_access_order_defaults.pp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  $_defaults = [
+    'expire',
+    'lockout'
+  ]
+
+  if $facts['sssd_version'] {
+    if versioncmp($facts['sssd_version'], '1.14.0') >= 0 {
+      $_result = [
+        'ppolicy',
+        'pwd_expire_policy_reject',
```
Comment (lnemsick-simp) [inline functions/ldap_access_order_defaults.pp]: Based on the descriptions of pwd_expire_policy_reject, pwd_expire_policy_warn, and pwd_expire_policy_renew, it only makes sense to me that one of these options is used in the ldap_access_order.  As it is written, pwd_expire_policy_reject will be applied.

Comment (trevor-vaughan) [reply functions/ldap_access_order_defaults.pp]: I'll drop it back to just `reject`

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-sssd/pipelines/42198816

**Pull request merged**